### DISCUSSION TITLE
Transitive @capacitor dependency not detected

### DISCUSSION THREAD

**Discussion by Maximvdw** [ionic-team/capacitor#6391] [Q&A]
Hey,

I have a wrapper (`MyWrapper`) for a capacitor plugin (@capacitor/motion) that has @capacitor/motion as a dependency. I want the @capacitor/motion plugin to be added when I add `MyWrapper` to the dependencies of my app, but whenever I run `npx cap sync` the plugin is not detected.

If I add `@capacitor/motion` to the package.json of my app, it works fine.

Is there a way so I can indicate that `MyWrapper` has a capacitor plugin as a dependency?

Kind regards,
Maxim

**Comment by tremorrisdev** [comment]
Hi there,

Not sure if you're still having this issue, but looking at how the CLI resolves package dependencies
https://github.com/ionic-team/capacitor/blob/b545938e42b14d65c7f5648a583377577ef28f7c/cli/src/plugin.ts#L116-L121
you can fix this by including `@capacitor/motion` in your packages as a dependency or devDependency. I personally went the devDependency route, as really, we only require the package at build (or dev) time.

I think this feels like an inappropriate fix as that plugin is definitely needed at runtime, but would otherwise be included *automagically* at build time if the CLI could analyze transitive dependencies. Perhaps `peerDependencies` would be more appropriate?
